src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -327,3 +316,38 @@ def get_config_actions(self) -> List[Tuple[str, str]]:
                 ),
             ),
         ]
+
+    @wrap_api_errors
+    def pre_register_submission(self, submission: "Submission", options: dict) -> None:
+        """
+        Create a Zaak, so that we can have a registration ID.
+
+        Note: The Rol, Status, the documents for the files uploaded by the user in the form (attachments) and the
+        confirmation report PDF will be added in the registration task (after the report has been generated).
+        """
+        zgw = ZgwConfig.get_solo()
+        zgw.apply_defaults_to(options)
+
+        zaak_data = apply_data_mapping(
+            submission, self.zaak_mapping, REGISTRATION_ATTRIBUTE
+        )
+
+        _create_zaak = partial(
+            create_zaak,
+            options,
+            payment_required=submission.payment_required,
+            existing_reference=submission.public_registration_reference,
+            **zaak_data,
+        )
```

Comment:
Hmm. So `execute_unless_result_exists` never calls this partial with any other arguments... Technically still partial application of the function, but do you think this is clearer?

```suggestion
        _create_zaak = lambda: create_zaak(
            options,
            payment_required=submission.payment_required,
            existing_reference=submission.public_registration_reference,
            **zaak_data,
        )
```

Reply:
Yes, maybe it is a bit clearer! Although I am not sure if there was any rationale for using the partial vs the lambda when this code was written :thinking:

Reply:
Oh yes. I hate that rule. It hurts readability.